### PR TITLE
Move helpers into `_utils`

### PR DESCRIPTION
**Pull request by lbittarello**
A little clean-up:

- `_expand_categorical_penalties` doesn't leverage the scope at all, so it doesn't need to be an inner function.
- `_is_contiguous` is self-standing, so it doesn't need to be a method.
- No reason to use `chain` anywhere.

### CODE REVIEW

**Review by MatthiasSchmidtblaicherQC — APPROVED** [review]
Thanks for spotting and taking care of this!

File: src/glum/_util.py [hunk 1]
```diff
@@ -114,6 +114,54 @@ def _add_missing_categories(
     return df
 
 
+def _expand_categorical_penalties(
+    penalty, X, drop_first, has_missing_category
+) -> np.ndarray:
+
+    # If `P1` or `P2` has the same shape as `X` before expanding
+    # categorical columns, we assume that the penalty at the location of
+    # categorical columns is the same for all levels.
```
Comment (lbittarello) [inline src/glum/_util.py:123]:
```suggestion
```

File: src/glum/_glm.py [hunk 2]
```diff
@@ -2679,6 +2675,7 @@ def _set_up_and_check_fit_args(
         P2 = self.P2
 
         copy_X = self._should_copy_X()
+        drop_first = getattr(self, "drop_first", False)
```
[the remark this answers is not in the document]
Comment (lbittarello) [reply src/glum/_glm.py:2678]: It is. :)

File: src/glum/_util.py [hunk 3]
```diff
@@ -114,6 +114,54 @@ def _add_missing_categories(
     return df
 
 
+def _expand_categorical_penalties(
+    penalty, X, drop_first, has_missing_category
+) -> np.ndarray:
+"""Determine penalty matrices`P1` or `P2` after expanding categorical columns.
```
Comment (MatthiasSchmidtblaicherQC) [inline src/glum/_util.py:120]:
```suggestion
"""Determine penalty matrices `P1` or `P2` after expanding categorical columns.
```

**Pull request merged**